### PR TITLE
Updated Camera2 example with CameraX API

### PR DESCRIPTION
**Pull request by anandwana001**
**What does this implement/fix? Explain your changes**
Refactor the Camera2 example with the new `CameraX` API.

**Does this close any currently open issues?**
No

**Any relevant logs, error output, bugreport etc?**

No

**Any other comments?**
- Do we need the Video feature of CameraX API?
- Do we need any modification in UI?

+ **Where has this been tested?**
Google Pixel 3a, physical device

+ **Device Information:**

+ **Android Version:**
Android 11

+ **Target Platform:**
32

+ **SDK Version:**
minimum - 21
compiled with - 32

+ **Configuration Information:**

+ **Misc:** 
Screenshot

### CODE REVIEW

**Comment by anandwana001** [comment]
@nisrulz PTAL

File: CameraX/app/src/main/java/github/nisrulz/example/cameraX/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+        if (allPermissionsGranted()) {
+            startCamera()
+        } else {
+            // Request for permissions mentioned in [REQUIRED_PERMISSIONS].
+            // This will invoke the [onRequestPermissionsResult]
+            ActivityCompat.requestPermissions(
+                this, REQUIRED_PERMISSIONS, REQUEST_CODE_PERMISSIONS
+            )
+        }
```
Comment (anandwana001) [inline CameraX/app/src/main/java/github/nisrulz/example/cameraX/MainActivity.kt:57]: As per the official document quote
>Ask for permissions in context, when the user starts to interact with the feature that requires it.

I am thinking to execute this only when a user tries to click on the capture button. 
WDYT?

File: CameraX/app/build.gradle [hunk 2]
```diff
@@ -22,18 +21,28 @@ android {
     }
 
     buildFeatures { viewBinding true }
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
```
Comment (nisrulz) [inline CameraX/app/build.gradle:28]: This is provided by the Kotlin Plugin, so this not needed anymore.

Comment (nisrulz) [reply CameraX/app/build.gradle:28]: Not a blocker though.

**Review by nisrulz — APPROVED** [review]
Tested it on a device. Everything works! Good work 🚀

**Pull request merged**